### DISCUSSION TITLE
Build got was canceled unexpectedly

### DISCUSSION THREAD

**Discussion by mittalyashu** [community/community#26100] [Actions]
<p>While running the build command, the job got cancalled for some unexpected reason.</p>
<p><a href="https://github.com/gitthermal/thermal/runs/202804644#step:5:414" rel="nofollow noopener">https://github.com/gitthermal/thermal/runs/202804644#step:5:414</a></p>
<p></p><div class="lightbox-wrapper"><a class="lightbox" href="https://aws1.discourse-cdn.com/github/original/2X/d/d2b3cfa84091486219a6932dbcf8192804a6c6ad.png" data-download-href="/uploads/short-url/u3XvLMsvUaClD839EqnZrE7fRvv.png?dl=1" title="chrome_AeElx6Avum.png" rel="nofollow noopener"><img src="https://user-images.githubusercontent.com/29014463/181095427-913ec344-38fc-48af-a711-179965c77919.png" alt="chrome_AeElx6Avum.png" data-base62-sha1="u3XvLMsvUaClD839EqnZrE7fRvv" width="690" height="243" data-small-upload="https://aws1.discourse-cdn.com/github/optimized/2X/d/d2b3cfa84091486219a6932dbcf8192804a6c6ad_2_10x10.png"><div class="meta">
<svg class="fa d-icon d-icon-far-image svg-icon" aria-hidden="true"><use xlink:href="#far-image"></use></svg><span class="filename">chrome_AeElx6Avum.png</span><span class="informations">956×337 18 KB</span><svg class="fa d-icon d-icon-discourse-expand svg-icon" aria-hidden="true"><use xlink:href="#discourse-expand"></use></svg>
</div></a></div>

**Comment by mittalyashu** [comment]
<p>This is the configuration for the GitHub action workflow I am using but it is not working.</p>
<pre><code>name: Greetings

on: issues

on:
  # Remove PRs from dependabot
  pull_request:
    branches:
      - '!dependabot/*'
</code></pre>
<p>Here’s the error message, I am getting in the Actions tab.</p>
<p></p><div class="lightbox-wrapper"><a class="lightbox" href="https://aws1.discourse-cdn.com/github/original/2X/d/d9ba36339336f2520c190c27c1becd8d85ba69cf.png" data-download-href="/uploads/short-url/v46zcSDWmTobKNq7q8OjQh0ZeJV.png?dl=1" title="chrome_1tpYoWg9xM.png" rel="nofollow noopener"><img src="https://user-images.githubusercontent.com/29014463/181095430-5d5274d5-6c21-4cfa-852b-573797f20d41.png" alt="chrome_1tpYoWg9xM.png" data-base62-sha1="v46zcSDWmTobKNq7q8OjQh0ZeJV" width="690" height="116" data-small-upload="https://aws1.discourse-cdn.com/github/optimized/2X/d/d9ba36339336f2520c190c27c1becd8d85ba69cf_2_10x10.png"><div class="meta">
<svg class="fa d-icon d-icon-far-image svg-icon" aria-hidden="true"><use xlink:href="#far-image"></use></svg><span class="filename">chrome_1tpYoWg9xM.png</span><span class="informations">1023×172 3.24 KB</span><svg class="fa d-icon d-icon-discourse-expand svg-icon" aria-hidden="true"><use xlink:href="#discourse-expand"></use></svg>
</div></a></div>

**Comment by AndreaGriffiths11** [comment]
<p>Thanks for being here <a class="mention" href="/mittalyashu">@mittalyashu</a>, I believe you got this going? If you have any other issues please feel free to post again. In the meantime, we’re looking forward to seeing you around!</p>

**Comment by mittalyashu** [comment]
<p>No, I am still facing this same issue.</p>

**Comment by AndreaGriffiths11** [comment]
<p>Apologies for the delay, taking a look at your action I feel it might be a syntax issue, see this image and give it a go, let us know if it works.</p>
<p><img src="https://user-images.githubusercontent.com/20666190/181095432-3c53c49c-8d8c-41fb-acac-2f8e32b647db.png" alt="Screen Shot 2019-09-11 at 3.22.05 PM.png" data-base62-sha1="dKDRf5U4ARTQmvLohzv9kQHPKuu" width="207" height="91"></p>

**Comment by andymckay** [comment]
<p>If you’d like listen to two events, such as <code>issues</code> and <code>pull_request</code> send this is a list to <code>on:</code> for example: <code>on: [issues, pull_request]</code>. Hope that helps.</p>

**Comment by mittalyashu** [comment]
<p>I want to ignore few of branches from pull requests, that’s why I have added.</p>
<pre><code>on:
  issues:
  # Remove PRs from dependabot
  pull_request:
    branches:
      - '!dependabot/*'
</code></pre>

**Comment by zaherg** [comment]
<p>You are using the matrix for your actions, and one thing about the matrix is that it will fail fast if one of the tests didn’t pass [<a href="https://help.github.com/en/articles/workflow-syntax-for-github-actions#jobsjob_idstrategyfail-fast" rel="nofollow noopener">ref</a>].</p>
<p>In your workflow <code>Linux build on node 12.x</code> didn’t pass which case everything else to cancel.</p>
<p>To solve this, you should use <code>fail-fast: false</code> within your <strong>strategy</strong></p>
<p>So your workflow file should be like:</p>
<pre><code>name: Test &amp; Build
on: push

jobs:
  Test:
    name: Test on node ${{ matrix.node_version }} and ${{ matrix.os }}
    runs-on: ${{ matrix.os }}
    strategy:
**fail-fast: false** matrix:
        node_version: [8.x, 10.x, 12.x]
        os: [ubuntu-latest, windows-latest, macOS-latest]

    steps:
      - name: Checkout
        uses: actions/checkout@v1

      - name: Set Node.js version ${{ matrix.node_version }}
        uses: actions/setup-node@v1
        with:
          version: ${{ matrix.node_version }}

      - name: Install packages
        run: npm install

      - name: Run lint
        run: npm run lint

  Linux:
    name: Linux build on node ${{ matrix.node_version }}
    runs-on: ubuntu-latest
    strategy:
**fail-fast: false** matrix:
        node_version: [8.x, 10.x, 12.x]

    steps:
      - name: Checkout
        uses: actions/checkout@v1

      - name: Set Node.js version ${{ matrix.node_version }}
        uses: actions/setup-node@v1
        with:
          version: ${{ matrix.node_version }}

      - name: Install packages
        run: npm install

      - name: Run build
        run: npm run build:linux
</code></pre>

**Comment by zaherg** [comment, marked as answer]
<p>I wrote a big answer but for some reason I cant find it so I’ll make it short.</p>
<p>Use <code>fail-fast</code> in your matrix <a href="https://help.github.com/en/articles/workflow-syntax-for-github-actions#jobsjob_idstrategyfail-fast" rel="nofollow noopener">https://help.github.com/en/articles/workflow-syntax-for-github-actions#jobsjob_idstrategyfail-fast</a></p>
<p>The build got canceled because one of your linux biuld failed, this one:</p>
<pre><code>Linux build on node 12.x
</code></pre>

**Comment by mittalyashu** [comment]
<p>Ah! I see.</p>
<p>I will fix it and give it a spin!</p>
